connect/runtime/src/test/java/org/apache/kafka/connect/util/KafkaBasedLogTest.java

```diff
@@ -115,10 +115,32 @@ public class KafkaBasedLogTest {
     private static final String TP1_VALUE_NEW = "VAL1_NEW";
 
     private Time time = new MockTime();
-    private KafkaBasedLog<String, String> store;
+
+    private class MockedKafkaBasedLog extends KafkaBasedLog<String, String> {
+        public MockedKafkaBasedLog(String topic,
+                                   Map<String, Object> producerConfigs,
+                                   Map<String, Object> consumerConfigs,
+                                   Supplier<TopicAdmin> topicAdminSupplier,
+                                   Callback<ConsumerRecord<String, String>> consumedCallback,
+                                   Time time,
+                                   Consumer<TopicAdmin> initializer) {
+            super(topic, producerConfigs, consumerConfigs, topicAdminSupplier, consumedCallback, time, initializer);
+        }
+
+        @Override
+        protected KafkaProducer<String, String> createProducer() {
+            return producer;
+        }
+
+        @Override
+        protected MockConsumer<String, String> createConsumer() {
+            return consumer;
+        }
+    }
+    private MockedKafkaBasedLog store;
 
     @Mock
-    private Runnable initializer;
+    private Consumer<TopicAdmin> initializer;
```

Comment:
As pointed out by @gharris1727 , the type of `initializer` had to be changed because it was in fact incorrect. I have no idea how this happened to work beforehand with EasyMock, but this appears to work alongside with changing `expectStart` to `verify(initializer).accept(any());` (since we have a `Supplier` and not a `Runnable`)

Reply:
The existing tests use a different (now-deprecated) constructor that accepts a `Runnable` initializer: https://github.com/apache/kafka/blob/dc18dd921c1e6f94f19212f45a4f851b1894dbdb/connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java#L125-L132

It should be fine to use the new constructor since it's trivially verifiable that the logic in this PR replicates the logic in the older constructor and we don't have to worry about a loss of coverage.